Release 20250123

mrghg opened the pull request:
Final tidy-up before the 20250123 release. This PR needs to be merged with #123 and #127, when they are complete.

- Added a code changelog
- Added a changelog for the AGAGE data
- Added documentation for the optical data processing
- Updated code and release version numbers

Still to do:

- [x] Update DOI when draft has been requested

mrghg commented:
Hi @joe-pitt, I updated the dvc file on Google drive for the proposed release. If you can remember how to access it, I'd be really grateful if you take a look (instructions in the Workflow document)

lukewestern commented:
I just switched onto the branch and ran the tests. I get some failures:
==================================================== short test summary info =====================================================
FAILED tests/test_formatting.py::test_cf_compliance - FileNotFoundError: Folder or zip archive /user/home/lw13938/agage-archive/data/agage_test/data-optical doesn't exist
FAILED tests/test_io.py::test_picarro - FileNotFoundError: Can't find file matching *-Picarro*_THD_ch4.nc in data/agage_test/data-optical
FAILED tests/test_io.py::test_picarro_inlet_grouping - FileNotFoundError: Can't find file matching *-Picarro*_TAC_ch4.nc in data/agage_test/data-optical
=========================================== 3 failed, 32 passed, 15 warnings in 18.91s ===========================================

I assume that data/agage_test/data-optical should be data/agage_test/data-optical-nc?

lukewestern commented:
Some more observations. Some may be for the data providers to sort out (i.e. data flagging issues). This is just for the combined and monthly-baseline files:

1) CFC-114 is being released for CMN in 2023 but shouldn't be (both in individual and monthly files)

2) CO monthly-basline is only released for MHD. Why is CGO (and TAC) missing here? It's released through the website. 

3) CHClCCl2 (TCE): I didn't think we released this because of data issues? I could be wrong.

4) CH3CCl3: MHD has lots of zero values in 2002. Either a mistake somewhere or should be flagged.

5) At TOB: There are lots of anomalously low points in 2019 that seem like they should be flagged to me in the combined files: HFC-143a, HFC-125, HCFC-142b.

6) CMN: There are zero values in the combined files for HCFC-22 in 2023. Also zero values in 2015 for HFC-245fa. Also a strange low point that should probably be flagged in 2015 for HCFC-141b.

7) JFJ: H-1301 has a lot of zero values in the early 2000s.

mrghg commented:
> I just switched onto the branch and ran the tests. I get some failures: ==================================================== short test summary info ===================================================== FAILED tests/test_formatting.py::test_cf_compliance - FileNotFoundError: Folder or zip archive /user/home/lw13938/agage-archive/data/agage_test/data-optical doesn't exist FAILED tests/test_io.py::test_picarro - FileNotFoundError: Can't find file matching _-Picarro__THD_ch4.nc in data/agage_test/data-optical FAILED tests/test_io.py::test_picarro_inlet_grouping - FileNotFoundError: Can't find file matching _-Picarro__TAC_ch4.nc in data/agage_test/data-optical =========================================== 3 failed, 32 passed, 15 warnings in 18.91s ===========================================
> 
> I assume that data/agage_test/data-optical should be data/agage_test/data-optical-nc?

Good spot. Fixed in config.py (you'll need to edit your config.yml)

joe-pitt commented:
> Some more observations. This is about the public release file and not the code itself. Some issues may be for the data providers to sort out (i.e. data flagging issues). This is just for the combined and monthly-baseline files:
> 
> 1. CFC-114 is being released for CMN in 2023 but shouldn't be (both in individual and monthly files). A lot of the earlier record (pre-2010) at CMN also looks like it has issues.
> 2. CO monthly-basline is only released for MHD. Why is CGO (and TAC) missing here? It's released through the website.
> 3. CHClCCl2 (TCE): I didn't think we released this because of data issues? I could be wrong.
> 4. CH3CCl3: MHD has lots of zero values in 2002. Either a mistake somewhere or should be flagged.
> 5. At TOB: There are lots of anomalously low points in 2019 that seem like they should be flagged to me in the combined files: HFC-143a, HFC-125, HCFC-142b.
> 6. CMN: There are zero values in the combined files for HCFC-22 in 2023. Also zero values in 2015 for HFC-245fa. Also a strange low point that should probably be flagged in 2015 for HCFC-141b.
> 7. JFJ: H-1301 has a lot of zero values in the early 2000s.

I think I've addressed the TAC one - we're not releasing the TAC MD/LGR data through AGAGE this time. I'll have a look at the rest either tomorrow (if I'm back from Jodrell Bank in time) or Thursday.

lukewestern commented:
Sorry for the confusion:

4) This wasn't MHD, it was CMN. At CMN there are zero values for CH3CCl3 in 2002.

joe-pitt commented:
> Some more observations. This is about the public release file and not the code itself. Some issues may be for the data providers to sort out (i.e. data flagging issues). This is just for the combined and monthly-baseline files:
> 
> 1. CFC-114 is being released for CMN in 2023 but shouldn't be (both in individual and monthly files). A lot of the earlier record (pre-2010) at CMN also looks like it has issues.

Good spot - just fixed the recent data. I've asked Jgor about the early data.
> 2. CO monthly-basline is only released for MHD. Why is CGO (and TAC) missing here? It's released through the website.

TAC is withheld for now. I asked Krum and he said we should withhold CGO CO because of a scale issue. It is marked to be withheld in Ray's ppt too.

> 3. CHClCCl2 (TCE): I didn't think we released this because of data issues? I could be wrong.

Definitely true for the Medusas - the data should only be there for the MHD and JFJ ADS. Is that what you see?
> 4. CH3CCl3: CMN has lots of zero values in 2002. Either a mistake somewhere or should be flagged.

> 5. At TOB: There are lots of anomalously low points in 2019 that seem like they should be flagged to me in the combined files: HFC-143a, HFC-125, HCFC-142b.

> 6. CMN: There are zero values in the combined files for HCFC-22 in 2023. Also zero values in 2015 for HFC-245fa. Also a strange low point that should probably be flagged in 2015 for HCFC-141b.

> 7. JFJ: H-1301 has a lot of zero values in the early 2000s.

I've emailed the relevant people about these issues. All are present in gccompare.

lukewestern commented:
One more thing. In the attributes for the combined files (below example for HCFC-22 monthly baseline for MHD), should the attributes contain an entry for each instrument indicated by number?
e.g., the intrument types in the variable are 3 and 4 (and -1 for NaN).
The attrs contains `instrument` and `instrument_1`. Should this be `instrument_3` and  `nstrument_4`?
```
site_code : MHD
network : agage
instrument_selection : Recommended instrument(s) selected and combined by AGAGE PIs
instrument : GCMS-Medusa
instrument_date : 2003-05-22
instrument_comment :
instrument_1 : GCMS-ADS
instrument_date_1 : 1998-02-16
instrument_comment_1 :
instrument_type : GCMS-ADS/GCMS-Medusa
```

lukewestern commented:
Yet another thing. I think there was some agreement that we should be releasing ZEP CH4? Which isn't included in this release (should maybe double confirm with Chris L. and others).

joe-pitt commented:
> Yet another thing. I think there was some agreement that we should be releasing ZEP CH4? Which isn't included in this release (should maybe double confirm with Chris L. and others).

Only the data processed through the ICOS system (not through GCWerks) is good to use in recent years. So we need to make a merged file include GCWerks data for the older years and ICOS data for the recent years. Ray Wang was looking into this.

mrghg commented:
> One more thing. In the attributes for the combined files (below example for HCFC-22 monthly baseline for MHD), should the attributes contain an entry for each instrument indicated by number? e.g., the intrument types in the variable are 3 and 4 (and -1 for NaN). The attrs contains `instrument` and `instrument_1`. Should this be `instrument_3` and `nstrument_4`?
> 
> ```
> site_code : MHD
> network : agage
> instrument_selection : Recommended instrument(s) selected and combined by AGAGE PIs
> instrument : GCMS-Medusa
> instrument_date : 2003-05-22
> instrument_comment :
> instrument_1 : GCMS-ADS
> instrument_date_1 : 1998-02-16
> instrument_comment_1 :
> instrument_type : GCMS-ADS/GCMS-Medusa
> ```

This is a bit of a known inconsistency, but I don't think I'll be able to fix it any time soon (the code that handles this is a bit of a mess).

The issue is that I'm using "instrument_type" to identify the broader instrument class (GCMD, GCMS-Medusa, etc.), but GCWerks outputs these attributes if there has been some identified change in the specific instrument (e.g., a change in the MS) WITHIN one instrument class. 

My code concatenates these fields, and then modifies the number so that they are sequential. So the numbering after "instrument_" doesn't have any relationship to the number for "instrument_type".

A workaround could be to re-label these attributes as "instrument_A", "instrument_B", etc. so that at least it doesn't look like the same thing as "instrument_type". Or, maybe we just drop the different instrument attributes and concatenate the text into three attributes ("instrument", "instrument_comment", "instrument_date"). Happy to hear any suggestions!

mrghg commented:
> > Yet another thing. I think there was some agreement that we should be releasing ZEP CH4? Which isn't included in this release (should maybe double confirm with Chris L. and others).
> 
> Only the data processed through the ICOS system (not through GCWerks) is good to use in recent years. So we need to make a merged file include GCWerks data for the older years and ICOS data for the recent years. Ray Wang was looking into this.

I guess we're skipping this for this release then? Also, isn't this something we should be adding into this repo, rather than having this extra step of Ray doing it? Not sure I fully understand.

joe-pitt commented:
> A workaround could be to re-label these attributes as "instrument_A", "instrument_B", etc. so that at least it doesn't look like the same thing as "instrument_type". Or, maybe we just drop the different instrument attributes and concatenate the text into three attributes ("instrument", "instrument_comment", "instrument_date"). Happy to hear any suggestions!

The _A, _B approach seems good, but are these variable names standardised within the agage-archive code or does it take whatever is in Peter's file? At the moment in GCWerks we have the freedom to specify "_1" or "_A"

joe-pitt commented:
> > > Yet another thing. I think there was some agreement that we should be releasing ZEP CH4? Which isn't included in this release (should maybe double confirm with Chris L. and others).
> > 
> > 
> > Only the data processed through the ICOS system (not through GCWerks) is good to use in recent years. So we need to make a merged file include GCWerks data for the older years and ICOS data for the recent years. Ray Wang was looking into this.
> 
> I guess we're skipping this for this release then? Also, isn't this something we should be adding into this repo, rather than having this extra step of Ray doing it? Not sure I fully understand.

Ray will need to rerun the pollution algorithm, so he will need the merge file in whatever format his pollution algorithm accepts

joe-pitt commented:
Had to tweak the cmn cfc fix from before - because the CMN medusa starts after the 113/114 cutoff date we actually need to exclude in the release csv or the code throws an error

joe-pitt commented:
One other thing to fix - inlet heights. Because the ADS data doesn't get reintegrated there are zero inlet heights where there shouldn't be for ZEP ADS and MHD ADS. Also RPB Medusa had a ports log error that has now been fixed, but Dickon wants us to stick with the original frozen data. Probably easiest if we manually edit the inlet height in Peter's .nc files to fix this. I can do that just before we do the final run (i.e. after we get the final set of files from Ray)

mrghg commented:
Now released: https://www-air.larc.nasa.gov/missions/agage/data/version-history/20250123
